### PR TITLE
Add section in docs for Operator API

### LINKED ISSUE #2115: Add API section to operator docs
Currently the API reference of the Operator CRDs is only documented in the CRD itself. 
Ideally we should a page in docs like [Prometheus Operator API](https://prometheus-operator.dev/docs/operator/api/) where all the possible parameters to configure the operator are displayed.

This could be automatized with this too: https://github.com/ahmetb/gen-crd-api-reference-docs

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

This PR introduces https://github.com/ahmetb/gen-crd-api-reference-docs tool to generate automatically
API documentation for Agent Operator. The new doc changes are generated when running `make generate-crd`.

This is how it looks:

Screenshot 2022-09-16 at 16-53-48 API Reference

#### Which issue(s) this PR fixes

Closes #2115

#### PR Checklist

- [ ] CHANGELOG updated
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: pkg/operator/apis/monitoring/doc.go
Comment (captncraig) [inline pkg/operator/apis/monitoring/doc.go:1]: may as well add a basic package level doc comment so this file has a reason to exist, other than the tool requires it.

**Review by captncraig — APPROVED** [review]

**Comment by captncraig** [comment]
I can't speak to the style or layout for the generated docs or how it fits into our general theme, but I do think this is a much needed addition.

**Comment by rfratto** [comment]
Oh my, this might not display very well for some people:

image

I'm not sure what we should do about that. We might have to wait for @jdbaldry to come back before we get some options here, since the horizontal scroll bar doesn't appear for me and the overflow content gets hidden.

**Comment by rfratto** [comment]
My one nit would be maybe we should call this the CRD reference instead of API reference so it doesn't get confused with the other API page.

But I think this is a great idea, we might just have to take it a little slow to figure out how we can fix the formatting.

**Comment by marctc** [comment]
@rfratto is this [site](https://prometheus-operator.dev/docs/operator/api/#monitoring.coreos.com/v1.PodMonitor) displayed for your properly?

**Comment by karengermond** [comment]
> Oh my, this might not display very well for some people:
> 
> image
> 
> I'm not sure what we should do about that. We might have to wait for @jdbaldry to come back before we get some options here, since the horizontal scroll bar doesn't appear for me and the overflow content gets hidden.

Yeah, hopefully that can be fixed.  This is great given that it's a generated file. It seems weird that the field types don't have their own column, but beggars can't be choosers when you're getting generated docs for free.

**Review by karengermond — APPROVED** [review]
LGTM

**Review by rfratto — APPROVED** [review]
The display/formatting may be a little weird on smaller browser windows, but I think it's your call for if you want to wait until it can be fixed or merge now and fix the display later.

File: docs/sources/operator/api.md
Comment (karengermond) [inline docs/sources/operator/api.md:77]: Should these brackets be here?  There are other places where they show up, too.

File: docs/sources/operator/api.md
Comment (karengermond) [inline docs/sources/operator/api.md:7]: Can we spell this title out as we do for the "Custom Resource Quickstart"? So  "Custom Resource Definition Reference" or "Custom Resource Reference" if that's OK?

File: docs/sources/operator/api.md
Comment (marctc) [reply docs/sources/operator/api.md:77]: This is automatically generated by the tool meaning that this section is composed by a list of `MetricsDeployment`

File: docs/sources/operator/api.md
Comment (marctc) [reply docs/sources/operator/api.md:7]: I'm not sure if I understand. Why Quickstart? It's an extensive list of all possible options to configure the CRD of the agent operator.

Comment (gloriasc) [reply docs/sources/operator/api.md:7]: I think she means to change the title to "Custom Resource Reference" rather than "CRD Reference" :)

Comment (marctc) [reply docs/sources/operator/api.md:7]: Got it, will do it

**Comment by jdbaldry** [comment]
I've not yet looked into the website styling of tables and how we can handle them more responsively. Would you like me to take this to the website squad and see what I can find out?

**Comment by rfratto** [comment]
> I've not yet looked into the website styling of tables and how we can handle them more responsively. Would you like me to take this to the website squad and see what I can find out?

Yes please :) Aside from tables not having horizontal scroll bars, I would also love if we had a little bit of control over the styling so code-formatted text looked consistent between rows rather than how it looks now:

image

**Comment by marctc** [comment]
I will merge this as it is and create a follow-up issue to fix visualizations issues

**Pull request merged**
